### PR TITLE
add helm chart values to configure global server side rate limiting

### PR DESCRIPTION
**Pull request by jmurret**
Changes proposed in this PR:
- add helm chart values for:
  - 
- add templating and bats tests to ensure that the the server-config-configmap.yml adds the configuration

How I've tested this PR:
- unit tests
- CI

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/values.yaml [hunk 1]
```diff
@@ -1131,6 +1131,41 @@ server:
     # @type: string
     caCert: null
 
+  # Settings for potentially limiting timeouts, rate limiting on clients as well
```
Comment (curtbushko) [inline charts/consul/values.yaml:1134]: Nit. Could you add a link to the rate limits configuration?

https://developer.hashicorp.com/consul/docs/agent/config/config-files#request_limits

File: charts/consul/values.yaml [hunk 2]
```diff
@@ -1131,6 +1131,41 @@ server:
...
+      mode: "disabled"
+      
+      # Setting that controls how frequently RPC, gRPC, and HTTP
+      # queries are allowed to happen. In any large enough time interval, rate
+      # limiter limits the rate to RequestLimitsReadRate tokens per second.
+      #
+      # See https://en.wikipedia.org/wiki/Token_bucket for more about token
+      # buckets.
+      # @type: integer
+      readRate: -1
```
Comment (curtbushko) [inline charts/consul/values.yaml:1158]: Why -1 instead of 100 which is the default?

**Review by curtbushko — COMMENTED** [review]
Looks good.  Just a couple comments below but nothing I would block on.

File: charts/consul/values.yaml (hunk 2 above)
Comment (jmurret) [reply charts/consul/values.yaml:1158]: those are the defaults: https://github.com/hashicorp/consul/blob/main/agent/config/default.go#L106-L107

This documentation is incorrect.  I'll open a PR to correct it.

File: charts/consul/values.yaml (hunk 1 above)
Comment (jmurret) [reply charts/consul/values.yaml:1134]: Great call.  Added here: https://github.com/hashicorp/consul-k8s/pull/2170/commits/7ab9a14a946e9a6916f5aed01755555c55241101#diff-9d190488582b6dfdeb91fd565c91a267e5dc1c1b925a174174aade2a503f5373R1141

File: charts/consul/values.yaml (hunk 2 above)
Comment (curtbushko) [reply charts/consul/values.yaml:1160]: Awesome! thanks1

**Review by curtbushko — APPROVED** [review]
LGTM!

**Review by dhiaayachi — COMMENTED** [review]
Thank you for putting this together so quickly @jmurret !

LGTM

File: .changelog/2170.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+```release-note:feature
```
Comment (david-yu) [inline .changelog/2170.txt:1]: I think you forgot a closing ```

**Pull request merged**